machine_learning/logistic_regression.py

```diff
@@ -31,13 +31,19 @@ def sigmoid_function(z):
 def cost_function(h, y):
     return (-y * np.log(h) - (1 - y) * np.log(1 - h)).mean()
 
+def log_likelihood(X, Y, weights):
+    scores = np.dot(features, weights)
+    ll = np.sum( target*scores - np.log(1 + np.exp(scores)) )
+    return ll
```

Comment:
Do not create the variable __ll__, just return the result.  The function is self explanatory.  Of course, a docstring, doctests, and type hints would not hurt.

Reply:
__return np.sum(target*scores - np.log(1 + np.exp(scores)))__